inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -239,7 +239,7 @@
         if self.sota_cmd == 'rollback':
             self.snap_num = setup_helper.get_snapper_snapshot_number()
         snapshot = self.factory.create_snapshotter(
-            self.sota_cmd, self.snap_num, self.proceed_without_rollback)
+            self.sota_cmd, self.snap_num, self.proceed_without_rollback, self._is_reboot_device())
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** The method execute has been updated to use the new variable name _reboot_device. Ensure that the method's logic correctly interprets the value of this variable, especially in conditions where a reboot decision is made.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            snapshot = self.factory.create_snapshotter(
                self.sota_cmd, self.snap_num, self.proceed_without_rollback, self._reboot_device)
```